src/resolvers-cassandra/Settings/mutations.js

```diff
@@ -3,21 +3,151 @@
- * @returns {Promise.<{name: string, properties: {targetBBox: number[], defaultZoomLevel: number, logo: string, title: string, defaultLocation: number[], storageConnectionString: string, featuresConnectionString: string, mapzenApiKey: string, fbToken: string, supportedLanguages: string[]}}>}
+ * @param {{input: {siteType: string, targetBbox: number[], defaultZoomLevel: number, logo: string, title: string, name: string, defaultLocation: number[], storageConnectionString: string, featuresConnectionString: string, mapzenApiKey: string, fbToken: string, supportedLanguages: string[]}}} args
+ * @returns {Promise}
  */
-function createOrReplaceSite(args, res) { // eslint-disable-line no-unused-vars
+function createSite(args, res) { // eslint-disable-line no-unused-vars
+  return new Promise((resolve, reject) => {
+    const siteType = args && args.input && args.input.siteType;
+    if(!siteType || !siteType.length) {
+      return reject(`siteType for sitename ${args.input.name} is not defined`);
+    }
+
+    cassandraConnector.executeQuery('SELECT * FROM fortis.sitesettings WHERE sitename = ? ALLOW FILTERING;', [args.input.name])
+      .then(rows => {
+        if (!rows || !rows.length) {
+          insertTopics(siteType)
+            .then(() => {
+              return cassandraConnector.executeBatchMutations([{
+                query: `INSERT INTO fortis.sitesettings (
+                  geofence,
+                  defaultzoom,
+                  logo,
+                  title,
+                  sitename,
+                  languages,
+                  insertion_time
+                ) VALUES (?,?,?,?,?,?,?,toTimestamp(now()))`,
+                params: [
+                  args.input.targetBbox,
+                  args.input.defaultZoomLevel,
+                  args.input.logo,
+                  args.input.title,
+                  args.input.name,
+                  args.input.supportedLanguages
+                ]
+              }]);
+            })
+            .then(() => { 
+              resolve({ //TODO: finalize these property values
+                name: args.input.name,
+                properties: {
+                  targetBbox: args.input.targetBbox,
+                  defaultZoomLevel: args.input.defaultZoomLevel,
+                  logo: args.input.logo,
+                  title: args.input.title,
+                  defaultLocation: [],
+                  storageConnectionString: '',
+                  featuresConnectionString: '',
+                  mapzenApiKey: '',
+                  fbToken: '',
+                  supportedLanguages:args.input.supportedLanguages,
+                  name: args.input.name
+                }
+              });
+            })
+            .catch(reject);
+        } 
+        if (rows.length == 1) return reject(`Site with sitename ${args.input.name} already exists.`);
```

Comment:
I have a feeling this `if` is meant to be an `else if` as part of what looks like a 3-element `if/else` chain. I say this because it looks like the `else` below will always be called alongside the insert above.